Index: asdf-0.4.0/lib/utils.sh
===================================================================
--- asdf-0.4.0.orig/lib/utils.sh
+++ asdf-0.4.0/lib/utils.sh
@@ -16,6 +16,14 @@ asdf_dir() {
   echo $ASDF_DIR
 }
 
+asdf_install_dir() {
+  if [ -z $ASDF_INSTALL_DIR ]; then
+    export ASDF_INSTALL_DIR=$(echo $(asdf_install_dir))
+  fi
+
+  echo $ASDF_INSTALL_DIR
+}
+
 asdf_repository_url() {
   echo "https://github.com/asdf-vm/asdf-plugins.git"
 }
@@ -24,13 +32,13 @@ get_install_path() {
   local plugin=$1
   local install_type=$2
   local version=$3
-  mkdir -p $(asdf_dir)/installs/${plugin}
+  mkdir -p $(asdf_install_dir)/${plugin}
 
   if [ $install_type = "version" ]
   then
-    echo $(asdf_dir)/installs/${plugin}/${version}
+    echo $(asdf_install_dir)/${plugin}/${version}
   else
-    echo $(asdf_dir)/installs/${plugin}/${install_type}-${version}
+    echo $(asdf_install_dir)/${plugin}/${install_type}-${version}
   fi
 }
 
@@ -38,7 +46,7 @@ list_installed_versions() {
   local plugin_name=$1
   local plugin_path=$(get_plugin_path $plugin_name)
 
-  local plugin_installs_path=$(asdf_dir)/installs/${plugin_name}
+  local plugin_installs_path=$(asdf_install_dir)/${plugin_name}
 
   if [ -d $plugin_installs_path ]; then
     for install in $(ls -d ${plugin_installs_path}/*/ 2>/dev/null); do
Index: asdf-0.4.0/lib/commands/plugin-remove.sh
===================================================================
--- asdf-0.4.0.orig/lib/commands/plugin-remove.sh
+++ asdf-0.4.0/lib/commands/plugin-remove.sh
@@ -5,7 +5,7 @@ plugin_remove_command() {
   local plugin_path=$(get_plugin_path $plugin_name)
 
   rm -rf $plugin_path
-  rm -rf $(asdf_dir)/installs/${plugin_name}
+  rm -rf $(asdf_install_dir)/${plugin_name}
 
   grep -l "asdf-plugin: ${plugin_name}" $(asdf_dir)/shims/* 2>/dev/null | xargs rm -f
 }
Index: asdf-0.4.0/lib/commands/reshim.sh
===================================================================
--- asdf-0.4.0.orig/lib/commands/reshim.sh
+++ asdf-0.4.0/lib/commands/reshim.sh
@@ -20,7 +20,7 @@ reshim_command() {
     generate_shims_for_version $plugin_name $full_version
   else
     # generate for all versions of the package
-    local plugin_installs_path=$(asdf_dir)/installs/${plugin_name}
+    local plugin_installs_path=$(asdf_install_dir)/${plugin_name}
 
     for install in ${plugin_installs_path}/*/; do
       local full_version_name=$(echo $(basename $install) | sed 's/ref\-/ref\:/')
