--- php-version-0.12.1/php-version.sh.orig	2015-12-22 04:06:22.360134420 +0100
+++ ./php-version-0.12.1/php-version.sh	2015-12-22 04:06:50.882134530 +0100
@@ -100,7 +100,7 @@
       _PHP_VERSIONS=()
       for _PHP_REPOSITORY in "${_PHP_REPOSITORIES[@]}"; do
         for _dir in $(find $(echo $_PHP_REPOSITORY) -maxdepth 1 -mindepth 1 -type d 2>/dev/null); do
-          _PHP_VERSIONS=("${_PHP_VERSIONS[@]}" "$($_dir/bin/php-config --version 2>/dev/null)")
+          _PHP_VERSIONS=("${_PHP_VERSIONS[@]}" "$($_dir/usr/bin/php-config --version 2>/dev/null)")
         done
       done
 
@@ -137,7 +137,7 @@
 
     for _PHP_REPOSITORY in "${_PHP_REPOSITORIES[@]}"; do
       for _dir in $(find $_PHP_REPOSITORY -maxdepth 1 -mindepth 1 -type d 2>/dev/null); do
-        _TARGET_VERSION_FUZZY=("${_TARGET_VERSION_FUZZY[@]}" "$($_dir/bin/php-config --version 2>/dev/null)")
+        _TARGET_VERSION_FUZZY=("${_TARGET_VERSION_FUZZY[@]}" "$($_dir/usr/bin/php-config --version 2>/dev/null)")
       done
     done
 
@@ -145,7 +145,7 @@
 
     for _PHP_REPOSITORY in "${_PHP_REPOSITORIES[@]}"; do
       for _dir in $(find $_PHP_REPOSITORY -maxdepth 1 -mindepth 1 -type d 2>/dev/null); do
-        _PHP_VERSION="$($_dir/bin/php-config --version 2>/dev/null)"
+        _PHP_VERSION="$($_dir/usr/bin/php-config --version 2>/dev/null)"
         if [[ -n "$_TARGET_VERSION_FUZZY" && "$_PHP_VERSION" == "$_TARGET_VERSION_FUZZY" ]]; then
           local _PHP_ROOT=$_dir
           break;
@@ -163,8 +163,8 @@
   # export current paths
   export PHPRC=""
   [[ -f $_PHP_ROOT/etc/php.ini ]] && export PHPRC=$_PHP_ROOT/etc/php.ini
-  [[ -d $_PHP_ROOT/bin  ]]        && export PATH="$_PHP_ROOT/bin:$PATH"
-  [[ -d $_PHP_ROOT/sbin ]]        && export PATH="$_PHP_ROOT/sbin:$PATH"
+  [[ -d $_PHP_ROOT/usr/bin  ]]        && export PATH="$_PHP_ROOT/usr/bin:$PATH"
+  [[ -d $_PHP_ROOT/usr/sbin ]]        && export PATH="$_PHP_ROOT/usr/sbin:$PATH"
 
   # use configured manpath if it exists, otherwise, use `$_PHP_ROOT/share/man`
   local _MANPATH=$(php-config --man-dir)
