diff -ur ../php56.orig/ext/mysqlnd/config9.m4 ./ext/mysqlnd/config9.m4
--- ../php56.orig/ext/mysqlnd/config9.m4	2014-10-01 11:17:38.000000000 +0200
+++ ./ext/mysqlnd/config9.m4	2015-12-21 01:48:17.044389851 +0100
@@ -34,7 +34,7 @@
   test -z "$PHP_OPENSSL" && PHP_OPENSSL=no
 
   if test "$PHP_OPENSSL" != "no" || test "$PHP_OPENSSL_DIR" != "no"; then
-    AC_CHECK_LIB(ssl, DSA_get_default_method, AC_DEFINE(HAVE_DSA_DEFAULT_METHOD, 1, [OpenSSL 0.9.7 or later]))
+    AC_CHECK_LIB(crypto, DSA_get_default_method, AC_DEFINE(HAVE_DSA_DEFAULT_METHOD, 1, [OpenSSL 0.9.7 or later]))
     AC_CHECK_LIB(crypto, X509_free, AC_DEFINE(HAVE_DSA_DEFAULT_METHOD, 1, [OpenSSL 0.9.7 or later]))
 
     PHP_SETUP_OPENSSL(MYSQLND_SHARED_LIBADD, [AC_DEFINE(MYSQLND_HAVE_SSL,1,[Enable mysqlnd code that uses OpenSSL directly])])
diff -ur ../php56.orig/ext/openssl/config0.m4 ./ext/openssl/config0.m4
--- ../php56.orig/ext/openssl/config0.m4	2014-10-01 11:17:38.000000000 +0200
+++ ./ext/openssl/config0.m4	2015-12-21 01:49:00.499843541 +0100
@@ -16,7 +16,7 @@
     PHP_SETUP_KERBEROS(OPENSSL_SHARED_LIBADD)
   fi
 
-  AC_CHECK_LIB(ssl, DSA_get_default_method, AC_DEFINE(HAVE_DSA_DEFAULT_METHOD, 1, [OpenSSL 0.9.7 or later]))
+  AC_CHECK_LIB(crypto, DSA_get_default_method, AC_DEFINE(HAVE_DSA_DEFAULT_METHOD, 1, [OpenSSL 0.9.7 or later]))
   AC_CHECK_LIB(crypto, X509_free, AC_DEFINE(HAVE_DSA_DEFAULT_METHOD, 1, [OpenSSL 0.9.7 or later]))
 
   PHP_SETUP_OPENSSL(OPENSSL_SHARED_LIBADD, 
