--- release/rtool.py.orig	2017-06-07 21:39:22.721942661 +0200
+++ release/rtool.py	2017-06-07 21:37:52.689944273 +0200
@@ -66,16 +66,16 @@
 ]
 
 
-def get_version() -> str:
+def get_version():
     return runpy.run_path(VERSION_FILE)["VERSION"]
 
 
-def git(args: str) -> str:
+def git(args):
     with chdir(ROOT_DIR):
         return subprocess.check_output(["git"] + shlex.split(args)).decode()
 
 
-def get_snapshot_version() -> str:
+def get_snapshot_version():
     last_tag, tag_dist, commit = git("describe --tags --long").strip().rsplit("-", 2)
     tag_dist = int(tag_dist)
     if tag_dist == 0:
@@ -89,7 +89,7 @@
         )
 
 
-def archive_name(bdist: str) -> str:
+def archive_name(bdist):
     platform_tag = {
         "Darwin": "osx",
         "Windows": "win32",
@@ -107,7 +107,7 @@
     )
 
 
-def wheel_name() -> str:
+def wheel_name():
     return "{project}-{version}-{py_version}-none-any.whl".format(
         project=PROJECT_NAME,
         version=get_version(),
@@ -128,7 +128,7 @@
 
 
 @contextlib.contextmanager
-def chdir(path: str):
+def chdir(path):
     old_dir = os.getcwd()
     os.chdir(path)
     yield
